src/Delivery/Console/WarmUpCacheCommand.php

```diff
@@ -43,11 +43,15 @@ protected function execute(InputInterface $input, OutputInterface $output)
     {
         $spaceId = $input->getArgument('space-id');
         $token = $input->getArgument('token');
-        $cacheDir = $input->getArgument('cache-dir');
+
+        $cachePoolFactoryClass = $input->getArgument('cache-item-pool-factory-class');
+        $cacheItemPoolFactory = new $cachePoolFactoryClass();
+        if (!$cacheItemPoolFactory instanceof CacheItemPoolFactoryInterface) {
+            throw new \InvalidArgumentException("Cache item pool factory class must implement \Contentful\Console\CacheItemPoolFactoryInterface");
+        }
 
         $client = new Client($token, $spaceId);
-        $cache = new FilesystemAdapter($spaceId, 0, $cacheDir);
-        $warmer = new CacheWarmer($client, $cache);
+        $warmer = new CacheWarmer($client, $cacheItemPoolFactory->getCacheItemPool());
```

Comment:
I would add a check here, to be sure that the result of `$cacheItemPoolFactory->getCacheItemPool()` actually is an instance of `CacheItemPoolInterface`.